### PR TITLE
Should also unlock on draft status

### PR DESCRIPTION
**Pull request by kneeyo1**

### CODE REVIEW

**Review by bukzor — APPROVED** [review]
_technically_ there should be a failing test to match this change

**Review by lynnagara — CHANGES_REQUESTED** [review]
I don't think DRAFT is a state in github

File: lib/tf_lock/tf_lock_acquire.py [hunk 1]
```diff
@@ -70,11 +70,11 @@ def tf_lock_acquire(root_module: Path) -> ExitCode:
 
 
 # TODO: deduplicate wrt spec.gh.pr.PR.is_closed
-def is_pr_closed(gh_url: str) -> bool:
+def is_pr_unlockable(gh_url: str) -> bool:
     status = sh.stdout(
         ("gh", "pr", "view", gh_url, "--json", "state", "--jq", ".state")
     )
-    return status in ("CLOSED", "MERGED")
+    return status in ("CLOSED", "MERGED", "DRAFT")
```
Comment (bukzor) [inline lib/tf_lock/tf_lock_acquire.py:77]: It's not that simple, unfortunately:

```suggestion
def is_pr_unlockable(gh_url: str) -> bool:
    status = sh.json(
        ("gh", "pr", "view", gh_url, "--json", "state,isDraft")
    )
    ...
```

File: spec/scenarios/orphaned_lock.py [hunk 2]
```diff
@@ -46,3 +46,36 @@ def test(
...
+
+        slices.assert_locked()
+
+    with (
+        tacos_demo.PR.opened_for_slices(
+            slices, test_name, demo, tacos_branch, branch=2
+        ) as pr2,
+    ):
+        # check that pr2 plans correctly
+        assert pr2.check("Terraform Plan").wait().success
```
[the remark this answers is not in the document]
Comment (kneeyo1) [reply spec/scenarios/orphaned_lock.py:81]: ah didn't catch that, thanks

**Review by lynnagara — APPROVED** [review]

**Pull request merged**